core/camel-core-processor/src/main/java/org/apache/camel/processor/MulticastProcessor.java

```diff
@@ -910,6 +910,7 @@ protected Integer getExchangeIndex(Exchange exchange) {
         for (Processor processor : processors) {
             // copy exchange, and do not share the unit of work
             Exchange copy = processorExchangeFactory.createCorrelatedCopy(exchange, false);
+            copy.adapt(ExtendedExchange.class).setTransacted(exchange.isTransacted());
```

Comment:
There would be other EIPs such as split etc that would need something similar. Maybe we can set the transacted in the createCorrelatedCopy (however wiretap EIP would need to set it to false as the tap exchange should not be transacted. 

So can you look at which other EIPs that use the createCorrelatedCopy API and build unit tests like you have for multicast.

Reply:
Thansk @davsclaus and I will take a look.